encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/ReflectiveSetter.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.encoders.reflective;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.lang.reflect.Type;
+
+interface ReflectiveSetter {
+  void set(@NonNull Object obj, @Nullable Object val);
+
+  @NonNull
+  Class<?> getFieldRawType();
+
+  @Nullable
+  Type getFieldGenericType();
+
+  boolean isDecodedInline();
```

Comment:
food fo thought(no immediate action required): it seems that it could be useful to just store the `inline` flag inside the field descriptor. This will help by:
* not needing a method here
* and furthermore would allow us to remove the `decodeInline()` method altogether making the api surface smaller and the implementation below simpler.

wdyt?